tests/rust/tcp-test/close/wait.rs

```diff
@@ -224,3 +227,119 @@ fn wait_after_async_close_connecting_socket(libos: &mut LibOS, remote: &SocketAd
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_ACCEPT && qr.qr_ret == 0 => {
+            anyhow::bail!("accept() should not succeed because remote should not be connecting")
+        },
+        Ok(_) => anyhow::bail!("wait() should return an error on accept() after close()"),
+        Err(e) => anyhow::bail!("wait() should not time out. {:?}", e),
+    }
+
+    Ok(())
+}
+
+fn wait_for_connect_after_issuing_async_close(libos: &mut LibOS, remote: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(wait_for_connect_after_issuing_async_close));
+
+    // Create a connecting socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    let qt: QToken = libos.connect(sockqd, *remote)?;
+    let mut connect_finished: bool = false;
+
+    // Poll once to ensure that the connect() co-routine runs.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Err(e) if e.errno == libc::ETIMEDOUT => {},
+        // Can only complete with ECONNREFUSED because remote does not exist.
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECONNREFUSED => {
+            connect_finished = true
+        },
+        // If connect() completes successfully, something has gone wrong.
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CONNECT && qr.qr_ret == 0 => {
+            anyhow::bail!("connect() should not succeed because remote does not exist")
+        },
+        Ok(_) => anyhow::bail!("wait() should not succeed with connect()"),
+        Err(_) => anyhow::bail!("wait() should timeout with connect()"),
+    }
+
+    let qt_close: QToken = libos.async_close(sockqd)?;
+
+    if !connect_finished {
+        // Poll again to check that the connect() co-routine completed with an error, either canceled or refused.
+        match libos.wait(qt, Some(Duration::from_micros(0))) {
+            Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECONNREFUSED => {},
+            Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECANCELED => {},
+            // If connect() completes successfully, something has gone wrong.
+            Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CONNECT && qr.qr_ret == 0 => {
+                anyhow::bail!("connect() should not succeed because remote does not exist")
+            },
+            Ok(_) => anyhow::bail!("wait() should return an error on connect() after async_close()"),
+            Err(_) => anyhow::bail!("wait() should not time out"),
+        }
+    }
+
+    // Poll once to ensure the async_close() coroutine runs and finishes the close.
+    match libos.wait(qt_close, Some(Duration::from_micros(0))) {
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE && qr.qr_ret == 0 => {},
+        Ok(_) => anyhow::bail!("wait() should succeed with async_close()"),
+        Err(e) => anyhow::bail!("wait() should succeed. {:?}", e),
+    }
+
```

Comment:
Don't we want to poll again the `connect()` queue token to ensure that it is cancelled?